src/torchcodec/decoders/_core/CMakeLists.txt

```diff
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.18)
 project(TorchCodec)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+add_compile_options(-Wall -Wextra -pedantic -Werror)
```

Comment:
I did a brief google investigation and it looks like `add_compile_options` is similar to setting `CMAKE_CXX_FLAGS` (as we do below line 7-8), with a different scope.
While [some sources](https://www.reddit.com/r/cpp_questions/comments/xdv88y/difference_between_cmake_ways_to_add_compile_flags/) suggest that `add_compile_options` is better scoped, I feel like we should stick to setting `CMAKE_CXX_FLAGS` here. Because a) this is what we already do, so it's best to only have one way to set flags, and b) we want to set the flags of the entire project anyway, for all targets, which is what `CMAKE_CXX_FLAGS` does.